src/current/v24.1/monitor-and-debug-changefeeds.md

```diff
@@ -133,25 +133,26 @@ changefeed_emitted_bytes{scope="vehicles"} 183557
-`backfill_count`   | Number of changefeeds currently executing a backfill ([schema change]({% link {{ page.version.version }}/changefeed-messages.md %}#schema-changes) or initial scan). | Changefeeds
-`sink_batch_hist_nanos` | Time messages spend batched in the sink buffer before being flushed and acknowledged. | Nanoseconds
-`flush_hist_nanos` | Time spent flushing messages across all changefeeds. | Nanoseconds
-`checkpoint_hist_nanos` | Time spent checkpointing changefeed progress. | Nanoseconds
-`error_retries` | Total retryable errors encountered by changefeeds. | Errors
-`backfill_pending_ranges` | Number of [ranges]({% link {{ page.version.version }}/architecture/overview.md %}#architecture-range) in an ongoing backfill that are yet to be fully emitted. | Ranges
-`message_size_hist` | Distribution in the size of emitted messages. | Bytes
+| Metric           |  Description | Unit | Type
+-------------------+--------------+------+--------------------------------------------
+`changefeed.admit_latency` | Difference between the event's MVCC timestamp and the time the event is put into the memory buffer. | Nanoseconds | Histogram
+`changefeed.aggregator_progress` | The earliest timestamp up to which any [aggregator]({% link {{ page.version.version }}/how-does-an-enterprise-changefeed-work.md %}) is guaranteed to have emitted all values for which it is responsible. **Note:** This metric may regress when a changefeed restarts due to a transient error. Consider tracking the `changefeed.checkpoint_progress` metric, which will not regress. | Timestamp | Gauge
+`changefeed.backfill_count` | Number of changefeeds currently executing a backfill ([schema change]({% link {{ page.version.version }}/changefeed-messages.md %}#schema-changes) or initial scan). | Changefeeds | Gauge
+`changefeed.backfill_pending_ranges` | Number of [ranges]({% link {{ page.version.version }}/architecture/overview.md %}#architecture-range) in an ongoing backfill that are yet to be fully emitted. | Ranges | Gauge
+`changefeed.checkpoint_hist_nanos` | Time spent checkpointing changefeed progress. | Nanoseconds | Histogram
+`changefeed.checkpoint_progress` | The earliest timestamp of any changefeed's persisted checkpoint (values prior to this timestamp will never need to be re-emitted). | Timestamp | Histogram
+`changefeed.commit_latency` | Difference between the event's MVCC timestamp and the time it is acknowledged by the [downstream sink]({% link {{ page.version.version }}/changefeed-sinks.md %}). If the sink is batching events, then the difference is between the oldest event and when the acknowledgment is recorded. | Nanoseconds | Histogram
+<span class="version-tag">New in v24.1:</span> `changefeed.emitted_batch_sizes` | Size of batches emitted to the sink by all changefeeds. | Messages | Histogram
```

Comment:
Line 145 is the new content in the PR.

Reply:
It might be clearer to describe this metric as the `number of messages emitted by all changefeeds`. Let’s wait for input from other reviewers before making any changes.

Reply:
Yes, ok! I wondered if this was the average number of messages in a batch across changefeeds as well?

Reply:
Looking at the original [pr](https://github.com/cockroachdb/cockroach/commit/b3c4adde1e570e223bdc80148c7fc4bac609e722), this metrics was introduced more for internal debugging use withbatching sinks. Maybe it's fine to leave the description as it is.

Reply:
Do you want to add `measured by number of messages` here as well?

Reply:
I think that the messages is captures by the unit type, but I'm ok with including it if we think this will lead to less confusion. We could also note that this metric includes resolved timestamps if the resolved options is enabled, which may be worth calling out because they always have a batch size of 1.